core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/elasticapi/ElasticExtensions.kt

```diff
@@ -145,3 +149,29 @@ suspend fun <C : ElasticsearchClient, T> C.suspendUntil(block: C.(ActionListener
             override fun onFailure(e: Exception) = cont.resumeWithException(e)
         })
     }
+
+/**
+ * Store a [ThreadContext] and restore a [ThreadContext] when switching to a new coroutine.
+ *
+ * @param threadContext - The context to store when switching to a coroutine.
+ * @param oldContext - The old context to restore upon completion of the coroutine.
+ */
+class ElasticThreadContextElement(
+    private val threadContext: ThreadContext,
+    private var oldContext: StoredContext
```

Comment:
* Rename `oldContext` to `initialContext`.
* `oldContext` can be initialized from `threadContext.newStoredContext(true)` and doesn't have to be passed in.